gnomad/utils/annotations.py

```diff
@@ -1046,31 +1046,45 @@ def missing_callstats_expr() -> hl.expr.StructExpression:
 
 
 def set_female_y_metrics_to_na_expr(
-    t: Union[hl.Table, hl.MatrixTable]
+    t: Union[hl.Table, hl.MatrixTable],
+    freq_expr: Union[hl.expr.ArrayExpression, str] = "freq",
+    freq_meta_expr: Union[hl.expr.ArrayExpression, str] = "freq_meta",
+    freq_index_dict_expr: Union[hl.expr.DictExpression, str] = "freq_index_dict",
 ) -> hl.expr.ArrayExpression:
     """
     Set Y-variant frequency callstats for female-specific metrics to missing structs.
 
-    .. note:: Requires freq, freq_meta, and freq_index_dict annotations to be present in Table or MatrixTable
+    :param t: Table or MatrixTable for which to adjust female metrics.
+    :param freq_expr: Array expression or string annotation name for the frequency
+        array. Default is "freq".
+    :param freq_meta_expr: Array expression or string annotation name for the frequency
+        metadata. Default is "freq_meta".
+    :param freq_index_dict_expr: Dict expression or string annotation name for the
+        frequency metadata index dictionary. Default is "freq_index_dict".
+    :return: Hail array expression to set female Y-variant metrics to missing values.
+    """
+    if isinstance(freq_expr, str):
+        freq_expr = t[freq_expr]
+    if isinstance(freq_meta_expr, str):
+        freq_meta_expr = t[freq_meta_expr]
+    if isinstance(freq_index_dict_expr, str):
+        freq_index_dict_expr = t[freq_index_dict_expr]
 
-    :param t: Table or MatrixTable for which to adjust female metrics
-    :return: Hail array expression to set female Y-variant metrics to missing values
-    """
     female_idx = hl.map(
```

Comment:
maybe we should make this 
```
    xx_idx = hl.map(
```
since we've been moving away from using gender terms for our sex assignments (though I guess maybe we should rename the function also, which is also a breaking change)

Reply:
I think this is a post v4 change, we can do an overhaul for that and population